catalyst_voices/apps/voices/lib/pages/campaign/details/widgets/campaign_management.dart

```diff
@@ -24,32 +24,37 @@ class _CampaignManagementState extends State<CampaignManagement> {
 
   @override
   Widget build(BuildContext context) {
-    final currentStatus = context.watch<CampaignBuilderCubit>().campaignStatus;
-    return Row(
-      children: [
-        VoicesOutlinedButton(
-          child: Text(context.l10n.campaignManagement),
-          onTap: () async {
-            final result =
-                await CampaignManagementDialog.show(context, currentStatus);
-            _handleDialogResult(result);
-          },
-        ),
-        _CampaignStatusIndicator(
-          campaignStatus: CampaignPublish.draft,
-          currentStatus: currentStatus,
-        ),
-        _CampaignStatusIndicator(
-          campaignStatus: CampaignPublish.published,
-          currentStatus: currentStatus,
-        ),
-      ],
+    return BlocSelector<CampaignBuilderCubit, CampaignBuilderState,
+        CampaignPublish?>(
+      selector: (state) => state.publish,
+      builder: (context, publish) {
+        return Row(
+          children: [
+            VoicesOutlinedButton(
+              child: Text(context.l10n.campaignManagement),
+              onTap: () async {
+                final result =
+                    await CampaignManagementDialog.show(context, publish);
+                _handleDialogResult(result);
```

Comment:
if this is async function it may take while which means context may be disposed at this points.

Consider making it `StatefulWidget` and extract calling `CampaignManagementDialog` into function or just check wheter context is mounted before calling  `_handleDialogResult`.